pytorch_lightning/trainer/trainer.py

```diff
@@ -2094,7 +2094,11 @@ def _results(self) -> Optional[ResultCollection]:
             return active_loop._results
 
     def _exit_gracefully_on_signal(self) -> None:
-        if _fault_tolerant_training() and self._terminate_gracefully:
+        if _fault_tolerant_training():
+            # the signal should be sent to rank 0
+            should_terminate_gracefully = self.training_type_plugin.broadcast(self._terminate_gracefully)
```

Comment:
shouldn't this use `reduce_boolean_decision`?

Reply:
Actually, no because the signal can be set to 1, many or all processes. Used reduce instead.